state-synchronizer/src/coordinator.rs

```diff
@@ -189,27 +202,65 @@ impl<T: ExecutorProxyTrait> SyncCoordinator<T> {
         }
     }
 
-    async fn request_sync(&mut self, request: SyncRequest) {
-        self.known_version = self
+    /// Sync up coordinator state with the local storage.
+    async fn sync_state_with_local_storage(&mut self) {
+        self.local_state = self
             .executor_proxy
-            .get_latest_version()
+            .get_local_storage_state()
             .await
-            .expect("[state sync] failed to fetch latest version from storage");
+            .expect("[state sync] failed to sync with local storage");
```

Comment:
we have `panic_handler` setup so it'll crash the app, but we need to figure out how to setup it for test

Reply:
Good question. I didn't change this behavior. I guess the current approach is that it's preferable to restart in case state synchronizer fails to talk with the storage, which probably makes sense... (although I'm also fine with just returning error)